airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -463,18 +467,10 @@ def generate_new_table_name(self, is_intermediate: bool, suffix: str) -> str:
                 # see alias in dbt: https://docs.getdbt.com/docs/building-a-dbt-project/building-models/using-custom-aliases/
                 pass
             pass
+        elif not self.parent:
+            new_table_name = get_table_name(self.name_transformer, "", new_table_name, suffix, self.json_path)
         else:
-            if suffix:
-                new_table_name = self.name_transformer.normalize_table_name(f"{table_name}_{suffix}")
-            if new_table_name in tables_registry:
-                # TODO handle collisions deterministically between intermediate tables and children
-                for i in range(1, 1000):
-                    if suffix:
-                        new_table_name = self.name_transformer.normalize_table_name(f"{table_name}_{i}_{suffix}")
-                    else:
-                        new_table_name = self.name_transformer.normalize_table_name(f"{table_name}_{i}")
-                    if new_table_name not in tables_registry:
-                        break
+            new_table_name = get_table_name(self.name_transformer, self.json_path[0], new_table_name, suffix, self.json_path)
```

Comment:
While keeping your logic in `get_table_name`, we could make it even easier for the user to follow through the multiple levels of nesting, WDYT of `"_".join(self.json_path)` instead of  `self.json_path[0]`?

The parent will be truncated to fit in the proper limits but we would expose as much JSON path as possible?

Reply:
done in https://github.com/airbytehq/airbyte/pull/2206/commits/64df1375c9c870409889994ac8907df7150ca2d9